src/v/cluster/tx_gateway_frontend.cc

```diff
@@ -258,9 +258,15 @@ ss::future<try_abort_reply> tx_gateway_frontend::do_try_abort(
                     try_abort_reply{.aborted = true, .ec = tx_errc::none});
```

Comment:
> tx coordinator & rm_stm interaction during a failure recovery
could lead to a deadlock. Making a lock expirable to resolve it

can we describe the deadlock that occurred, in the commit message?

Reply:
done